Sources/Starknet/Data/Numbers/Felt.swift

```diff
@@ -28,6 +28,21 @@ public struct Felt: NumAsHexProtocol {
         self.value = value < Felt.prime ? value : Felt.prime - 1
     }
 
+    /// Initializes a new Felt instance from a signed integer.
+    ///
+    /// Calculated as `fromSigned mod Felt.prime`.
+    ///
+    /// - Parameter fromSigned: The signed integer to convert to Felt. The integer must be greater than `-Felt.prime` and less than `Felt.prime`.
+    public init?(fromSigned exactly: some BinaryInteger) {
+        let value = BigInt(exactly: exactly)
+
+        guard let value, abs(value) < Felt.prime else {
+            return nil
```

Comment:
should we return nil, or maybe rather throw an error?

Reply:
I am generally in favor of throwing errors, but since all the other initializers are failable, I decided to not break the established convention. 

We can add an issue though and determine whether we want to change this convention separately.

Reply:
I see. We can create an issue, but I doubt this will have high priority

Reply:
#183